RELEASE.md

```diff
@@ -87,24 +90,29 @@ you can skip the dependency update or only update select dependencies. In such a
 case, you have to create an issue or pull request in the GitHub project for
 later follow-up.
 
-#### Updating Go dependencies
+#### Manually updating Go dependencies
```

Comment:
Perhaps add a note along the lines: This is usually only required for dependencies not following the Go Modules semver semantics.

Reply:
I added a note about this in the dependabot section, but I'll add it here as well.